docs/source/scilla-in-depth.rst

```diff
@@ -1081,13 +1081,15 @@ below, ``Any`` can be of type ``IntX``, ``UintX``, ``String``, ``ByStr20`` or
   The return type is ``Option String``. On success, ``Some addr`` is returned and on invalid inputs ``None`` is returned.
 
 - ``builtin alt_bn128_G1_add p1 p2``. The builtin takes two points ``p1``, ``p2`` on the ``alt_bn128`` curve and returns 
-  the sum of the points in the underlying group G1. The input points and the result point are each a ``Pair {Bystr32 ByStr32}``.
+  the sum of the points in the underlying group G1. The input points and the result option of a point are each a ``Some (Pair (Bystr32) (ByStr32))``.
   Each scalar component ``ByStr32`` of a point is a big-endian encoded number.
   Also see https://github.com/ethereum/EIPs/blob/master/EIPS/eip-196.md
 
 - ``builtin alt_bn128_G1_mul p s``. The builtin takes a point ``p`` on the ``alt_bn128`` curve (as described previously),
-  and a scalar ``ByStr32`` value ``s`` and returns the sum of the point ``p`` taken ``s`` times. The result is a point
-  on the curve.
+  and a scalar ``ByStr32`` value ``s`` and returns the sum of the point ``p`` taken ``s`` times. The result is ``Option (Pair (ByStr32) (ByStr32))``.
```

Comment:
We should say something about the cases when the result is `Some ...` and `None`.